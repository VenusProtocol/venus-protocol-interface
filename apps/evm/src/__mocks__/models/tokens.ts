import bnbLogo from 'packages/tokens/img/bnb.svg';
import busdLogo from 'packages/tokens/img/busd.svg';
import ethLogo from 'packages/tokens/img/eth.svg';
import lisUsdLogo from 'packages/tokens/img/lisUSD.png';
import lunaLogo from 'packages/tokens/img/luna.svg';
import usdcLogo from 'packages/tokens/img/usdc.svg';
import usdtLogo from 'packages/tokens/img/usdt.svg';
import ustLogo from 'packages/tokens/img/ust.svg';
import vaiLogo from 'packages/tokens/img/vai.svg';
import vrtLogo from 'packages/tokens/img/vrt.svg';
import wbnbLogo from 'packages/tokens/img/wbnb.svg';
import xvsLogo from 'packages/tokens/img/xvs.svg';
import { Token } from 'types';

export const xvs: Token = {
  address: '0xB9e0E753630434d7863528cc73CB7AC638a7c8ff',
  decimals: 18,
  symbol: 'XVS',
  asset: xvsLogo,
};

export const vai: Token = {
  address: '0x5fFbE5302BadED40941A403228E6AD03f93752d9',
  decimals: 18,
  symbol: 'VAI',
  asset: vaiLogo,
};

export const vrt: Token = {
  address: '0x331F639B4F3CF6E391CD244e0b5027C5968Ec325',
  decimals: 18,
  symbol: 'VRT',
  asset: vrtLogo,
};

export const bnb: Token = {
  address: '0x0000000000000000000000000000000000000000',
  isNative: true,
  decimals: 18,
  symbol: 'BNB',
  asset: bnbLogo,
};

export const usdc: Token = {
  address: '0x16227D60f7a0e586C66B005219dfc887D13C9531',
  decimals: 6,
  symbol: 'USDC',
  asset: usdcLogo,
};

export const usdt: Token = {
  address: '0xA11c8D9DC9b66E209Ef60F0C8D969D3CD988782c',
  decimals: 6,
  symbol: 'USDT',
  asset: usdtLogo,
};

export const busd: Token = {
  address: '0x8301F2213c0eeD49a7E28Ae4c3e91722919B8B47',
  decimals: 18,
  symbol: 'BUSD',
  asset: busdLogo,
};

export const ust: Token = {
  address: '0x5A79efD958432E72211ee73D5DDFa9bc8f248b5F',
  decimals: 18,
  symbol: 'UST',
  asset: ustLogo,
};

export const luna: Token = {
  address: '0xf36160EC62E3B191EA375dadfe465E8Fa1F8CabB',
  decimals: 6,
  symbol: 'LUNA',
  asset: lunaLogo,
};

export const wbnb: Token = {
  symbol: 'WBNB',
  decimals: 18,
  address: '0xae13d989daC2f0dEbFf460aC112a837C89BAa7cd',
  asset: wbnbLogo,
};

export const eth: Token = {
  address: '0x98f7A83361F7Ac8765CcEBAB1425da6b341958a7',
  decimals: 18,
  symbol: 'ETH',
  asset: ethLogo,
};

export const lisUsd: Token = {
  address: '0xe73774DfCD551BF75650772dC2cC56a2B6323453',
  decimals: 18,
  symbol: 'lisUSD',
  asset: lisUsdLogo,
};

export default [xvs, bnb, usdc, usdt, busd, ust, luna, vai, wbnb, eth, vrt, lisUsd];
