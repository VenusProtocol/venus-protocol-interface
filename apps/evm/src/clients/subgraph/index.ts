export * from './queries/isolatedPools/getIsolatedPoolParticipantsCount';
export * from './queries/governance/getProposals';
