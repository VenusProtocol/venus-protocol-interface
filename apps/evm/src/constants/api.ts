import type { Network } from 'types';

export const apiUrls: Record<Network, string> = {
  testnet: 'https://testnetapi.venus.io',
  mainnet: 'https://api.venus.io',
  'mainnet-preview': 'https://api-preview.venus.io',
};
