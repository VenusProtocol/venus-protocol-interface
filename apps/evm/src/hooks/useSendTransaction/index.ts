import { ContractReceipt, ContractTransaction } from 'ethers';
import { MutationObserverOptions, useMutation } from 'react-query';

import FunctionKey from 'constants/functionKey';
import { TransactionType } from 'types';

import { CONFIRMATIONS, useTrackTransaction } from './useTrackTransaction';

export interface UseSendTransactionOptions<TMutateInput extends Record<string, unknown> | void>
  extends MutationObserverOptions<unknown, Error, TMutateInput> {
  waitForConfirmation?: boolean;
}

export interface UseSendTransactionInput<TMutateInput extends Record<string, unknown> | void> {
  fn: (input: TMutateInput) => Promise<ContractTransaction>;
  fnKey: FunctionKey | [FunctionKey, ...unknown[]];
  transactionType?: TransactionType;
  onConfirmed?: (input: {
    transaction: ContractTransaction;
    transactionReceipt: ContractReceipt;
    input: TMutateInput;
  }) => Promise<unknown> | unknown;
  onReverted?: (input: {
    transaction: ContractTransaction;
    input: TMutateInput;
  }) => Promise<unknown> | unknown;
  options?: UseSendTransactionOptions<TMutateInput>;
}

export const useSendTransaction = <TMutateInput extends Record<string, unknown> | void>({
  fn,
  fnKey,
  transactionType,
  onConfirmed,
  onReverted,
  options,
}: UseSendTransactionInput<TMutateInput>) => {
  const trackTransaction = useTrackTransaction({ transactionType });

  return useMutation(
    fnKey,
    async input => {
      // Send transaction
      const transaction = await fn(input);

      // Track transaction's progress in the background
      trackTransaction({
        transaction,
        onConfirmed: onConfirmedInput => onConfirmed?.({ ...onConfirmedInput, input }),
        onReverted: onRevertedInput => onReverted?.({ ...onRevertedInput, input }),
      });

      if (options?.waitForConfirmation) {
        // Only return when transaction has been confirmed
        await transaction.wait(CONFIRMATIONS);
      }
    },
    options,
  );
};
