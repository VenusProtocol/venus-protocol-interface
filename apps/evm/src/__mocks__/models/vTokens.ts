import type { VToken } from 'types';

import { bnb, busd, eth, lisUsd, luna, usdc, usdt, ust, wbnb, weth, xvs } from './tokens';

export const vXvs: VToken = {
  address: '0x6d6F697e34145Bb95c54E77482d97cc261Dc237E',
  decimals: 8,
  symbol: 'vXVS',
  underlyingToken: xvs,
};

export const vBnb: VToken = {
  address: '0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c',
  decimals: 8,
  symbol: 'vBNB',
  underlyingToken: bnb,
};

export const vUsdc: VToken = {
  address: '0xD5C4C2e2facBEB59D0216D0595d63FcDc6F9A1a7',
  decimals: 8,
  symbol: 'vUSDC',
  underlyingToken: usdc,
};

export const vUsdtCorePool: VToken = {
  address: '0xb7526572FFE56AB9D7489838Bf2E18e3323b441A',
  decimals: 8,
  symbol: 'vUSDT',
  underlyingToken: usdt,
};

export const vUsdtDeFi: VToken = {
  address: '0x80CC30811e362aC9aB857C3d7875CbcCc0b65750',
  decimals: 8,
  symbol: 'vUSDT_DeFi',
  underlyingToken: usdt,
};

export const vUsdtGameFi: VToken = {
  address: '0x0bFE4e0B8A2a096A27e5B18b078d25be57C08634',
  decimals: 8,
  symbol: 'vUSDT_GameFi',
  underlyingToken: usdt,
};

export const vUsdtLiquidStakedBnb: VToken = {
  address: '0x2197d02cC9cd1ad51317A0a85A656a0c82383A7c',
  decimals: 8,
  symbol: 'vUSDT_Liquid_Staked_BNB',
  underlyingToken: usdt,
};

export const vUsdtMeme: VToken = {
  address: '0x3AF2bE7AbEF0f840b196D99d79F4B803a5dB14a1',
  decimals: 8,
  symbol: 'vUSDT_Meme',
  underlyingToken: usdt,
};

export const vUsdtStablecoins: VToken = {
  address: '0x3338988d0beb4419Acb8fE624218754053362D06',
  decimals: 8,
  symbol: 'vUSDT_Stablecoins',
  underlyingToken: usdt,
};

export const vUsdtTron: VToken = {
  address: '0x712774CBFFCBD60e9825871CcEFF2F917442b2c3',
  decimals: 8,
  symbol: 'vUSDT_Tron',
  underlyingToken: usdt,
};

export const vBusdCorePool: VToken = {
  address: '0x08e0A5575De71037aE36AbfAfb516595fE68e5e4',
  decimals: 8,
  symbol: 'vBUSD',
  underlyingToken: busd,
};

export const vBusdStablecoins: VToken = {
  address: '0x8c8A1a0b6e1cb8058037F7bF24de6b79Aca5B7B0',
  decimals: 8,
  symbol: 'vBUSD',
  underlyingToken: busd,
};

export const vUst: VToken = {
  address: '0xF206af85BC2761c4F876d27Bd474681CfB335EfA',
  decimals: 8,
  symbol: 'vUST',
  underlyingToken: ust,
};

export const vLuna: VToken = {
  address: '0x9C3015191d39cF1930F92EB7e7BCbd020bCA286a',
  decimals: 8,
  symbol: 'vLUNA',
  underlyingToken: luna,
};

export const vWeth: VToken = {
  address: '0xc2931B1fEa69b6D6dA65a50363A8D75d285e4da9',
  decimals: 8,
  symbol: 'vWETH',
  underlyingToken: weth,
};

export const vEth: VToken = {
  address: '0x86f8DfB7CA84455174EE9C3edd94867b51Da46BD',
  decimals: 8,
  symbol: 'vETH',
  underlyingToken: eth,
};

export const vLisUSD: VToken = {
  address: '0x170d3b2da05cc2124334240fB34ad1359e34C562',
  decimals: 8,
  symbol: 'vlisUSD',
  underlyingToken: lisUsd,
};

export const vWbnb: VToken = {
  address: '0x231dED0Dfc99634e52EE1a1329586bc970d773b3',
  decimals: 8,
  symbol: 'vWBNB',
  underlyingToken: wbnb,
};

export default [
  vXvs,
  vBnb,
  vUsdc,
  vUsdtCorePool,
  vUsdtDeFi,
  vUsdtGameFi,
  vUsdtLiquidStakedBnb,
  vUsdtMeme,
  vUsdtStablecoins,
  vUsdtTron,
  vBusdCorePool,
  vBusdStablecoins,
  vUst,
  vLuna,
  vEth,
  vLisUSD,
  vWbnb,
];
