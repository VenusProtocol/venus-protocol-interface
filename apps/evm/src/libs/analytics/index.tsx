import { PostHogProvider, usePostHog } from 'posthog-js/react';
import { useEffect } from 'react';

import config from 'config';
import { version as APP_VERSION } from 'constants/version';
import { useAccountAddress, useChainId } from 'libs/wallet';

const UserIdentifier: React.FC = () => {
  const { accountAddress } = useAccountAddress();
  const { chainId } = useChainId();
  const posthog = usePostHog();

  // Identify user by their account address along with the network chain they are on
  useEffect(() => {
    if (accountAddress) {
      posthog?.identify(accountAddress, { chainId });
    } else {
      posthog?.reset();
    }
  }, [accountAddress, chainId, posthog]);

  return null;
};

export interface AnalyticProviderProps {
  children?: React.ReactNode;
}

export const AnalyticProvider: React.FC<AnalyticProviderProps> = ({ children }) => (
  <PostHogProvider
    apiKey={config.posthog.apiKey}
    options={{
      api_host: config.posthog.hostUrl,
      persistence: 'memory',
      name: APP_VERSION,
    }}
  >
    <UserIdentifier />

    {children}
  </PostHogProvider>
);

export * from './useAnalytics';
export { default as useAnalytics } from './useAnalytics';
