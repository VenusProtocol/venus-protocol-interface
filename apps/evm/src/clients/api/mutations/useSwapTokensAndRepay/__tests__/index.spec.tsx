import BigNumber from 'bignumber.js';
import { queryClient } from 'clients/api';
import { useSendTransaction } from 'hooks/useSendTransaction';
import { useAnalytics } from 'libs/analytics';
import { getSwapRouterContractAddress } from 'libs/contracts';
import { renderHook } from 'testUtils/render';
import type { Mock } from 'vitest';
import { useSwapTokensAndRepay } from '..';

import fakeAccountAddress, {
  altAddress as fakePoolComptrollerContractAddress,
} from '__mocks__/models/address';
import { bnb, xvs } from '__mocks__/models/tokens';
import { vBnb } from '__mocks__/models/vTokens';

vi.mock('libs/analytics');
vi.mock('libs/contracts');
vi.mock('utilities/generateTransactionDeadline');

const fakeAmountMantissa = new BigNumber('10000000000000000');

const fakeSwap = {
  direction: 'exactAmountIn' as const,
  fromToken: xvs,
  toToken: bnb,
  fromTokenAmountSoldMantissa: fakeAmountMantissa,
  minimumToTokenAmountReceivedMantissa: fakeAmountMantissa,
  expectedToTokenAmountReceivedMantissa: fakeAmountMantissa,
  exchangeRate: new BigNumber('1'),
  priceImpactPercentage: 0,
  routePath: [xvs.address, bnb.address],
};

const fakeInput = {
  swap: fakeSwap,
  vToken: vBnb,
  repayFullLoan: false,
  poolComptrollerContractAddress: fakePoolComptrollerContractAddress,
  poolName: 'Fake Pool',
};

const fakeOptions = {
  gasless: true,
  waitForConfirmation: true,
};

const mockCaptureAnalyticEvent = vi.fn();

describe('useSwapTokensAndRepay', () => {
  beforeEach(() => {
    (useAnalytics as Mock).mockImplementation(() => ({
      captureAnalyticEvent: mockCaptureAnalyticEvent,
    }));

    (getSwapRouterContractAddress as Mock).mockImplementation(
      () => 'fakeSwapRouterContractAddress',
    );
  });

  it('calls useSendTransaction with the correct parameters for swapping tokens and repaying', async () => {
    renderHook(() => useSwapTokensAndRepay(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(fakeInput)).toMatchInlineSnapshot(
      {
        abi: expect.any(Array),
      },
      `
      {
        "abi": Any<Array>,
        "address": "fakeSwapRouterContractAddress",
        "args": [
          10000000000000000n,
          10000000000000000n,
          [
            "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
            "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
          ],
          1747386407n,
        ],
        "functionName": "swapExactTokensForBNBAndRepay",
      }
    `,
    );

    onConfirmed({ input: fakeInput });

    expect(mockCaptureAnalyticEvent).toHaveBeenCalledTimes(1);
    expect(mockCaptureAnalyticEvent.mock.calls[0]).toMatchInlineSnapshot(`
      [
        "Tokens swapped and repaid",
        {
          "exchangeRate": 1,
          "fromTokenAmountTokens": 0.01,
          "fromTokenSymbol": "XVS",
          "poolName": "Fake Pool",
          "priceImpactPercentage": 0,
          "repaidFullLoan": false,
          "slippageTolerancePercentage": 0.5,
          "toTokenAmountTokens": 0.01,
          "toTokenSymbol": "BNB",
        },
      ]
    `);

    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('calls useSendTransaction with the correct parameters for swapping tokens and repaying full loan', async () => {
    renderHook(() => useSwapTokensAndRepay(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    const repayFullLoanInput = {
      ...fakeInput,
      repayFullLoan: true,
      swap: {
        ...fakeSwap,
        direction: 'exactAmountOut' as const,
        toTokenAmountReceivedMantissa: fakeAmountMantissa,
        maximumFromTokenAmountSoldMantissa: fakeAmountMantissa,
        expectedFromTokenAmountSoldMantissa: fakeAmountMantissa,
      },
    };

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(repayFullLoanInput)).toMatchInlineSnapshot(
      {
        abi: expect.any(Array),
      },
      `
      {
        "abi": Any<Array>,
        "address": "fakeSwapRouterContractAddress",
        "args": [
          10000000000000000n,
          [
            "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
            "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
          ],
          1747386407n,
        ],
        "functionName": "swapTokensForFullBNBDebtAndRepay",
      }
    `,
    );

    onConfirmed({ input: fakeInput });

    expect(mockCaptureAnalyticEvent).toHaveBeenCalledTimes(1);
    expect(mockCaptureAnalyticEvent.mock.calls[0]).toMatchInlineSnapshot(`
      [
        "Tokens swapped and repaid",
        {
          "exchangeRate": 1,
          "fromTokenAmountTokens": 0.01,
          "fromTokenSymbol": "XVS",
          "poolName": "Fake Pool",
          "priceImpactPercentage": 0,
          "repaidFullLoan": false,
          "slippageTolerancePercentage": 0.5,
          "toTokenAmountTokens": 0.01,
          "toTokenSymbol": "BNB",
        },
      ]
    `);

    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('calls useSendTransaction with the correct parameters for swapping BNB and repaying tokens', async () => {
    renderHook(() => useSwapTokensAndRepay(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    const swapBnbInput = {
      ...fakeInput,
      swap: {
        ...fakeSwap,
        fromToken: bnb,
        toToken: xvs,
        fromTokenAmountSoldMantissa: fakeAmountMantissa,
        minimumToTokenAmountReceivedMantissa: fakeAmountMantissa,
      },
    };

    expect(useSendTransaction).toHaveBeenCalledWith({
      fn: expect.any(Function),
      onConfirmed: expect.any(Function),
      options: fakeOptions,
    });

    const { fn, onConfirmed } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(await fn(swapBnbInput)).toMatchInlineSnapshot(
      {
        abi: expect.any(Array),
      },
      `
      {
        "abi": Any<Array>,
        "address": "fakeSwapRouterContractAddress",
        "args": [
          "0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c",
          10000000000000000n,
          [
            "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
            "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
          ],
          1747386407n,
        ],
        "functionName": "swapBNBForExactTokensAndRepay",
        "value": 10000000000000000n,
      }
    `,
    );

    onConfirmed({ input: fakeInput });

    expect(mockCaptureAnalyticEvent).toHaveBeenCalledTimes(1);
    expect(mockCaptureAnalyticEvent.mock.calls[0]).toMatchInlineSnapshot(`
      [
        "Tokens swapped and repaid",
        {
          "exchangeRate": 1,
          "fromTokenAmountTokens": 0.01,
          "fromTokenSymbol": "XVS",
          "poolName": "Fake Pool",
          "priceImpactPercentage": 0,
          "repaidFullLoan": false,
          "slippageTolerancePercentage": 0.5,
          "toTokenAmountTokens": 0.01,
          "toTokenSymbol": "BNB",
        },
      ]
    `);

    expect((queryClient.invalidateQueries as Mock).mock.calls).toMatchSnapshot();
  });

  it('throws when SwapRouter contract address is not available', async () => {
    (getSwapRouterContractAddress as Mock).mockImplementation(() => undefined);

    renderHook(() => useSwapTokensAndRepay(fakeOptions), {
      accountAddress: fakeAccountAddress,
    });

    const { fn } = (useSendTransaction as Mock).mock.calls[0][0];

    expect(async () => fn(fakeInput)).rejects.toThrow('somethingWentWrong');
  });
});
