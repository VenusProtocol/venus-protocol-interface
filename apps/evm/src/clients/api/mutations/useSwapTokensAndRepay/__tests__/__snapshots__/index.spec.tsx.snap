// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useSwapTokensAndRepay > calls useSendTransaction with the correct parameters for swapping BNB and repaying tokens 3`] = `
[
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_ALLOWANCE",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "spenderAddress": "0xfakeSwapRouterContractAddress",
          "tokenAddress": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_BALANCES",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCES_ALL",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_POOLS",
      ],
    },
  ],
]
`;

exports[`useSwapTokensAndRepay > calls useSendTransaction with the correct parameters for swapping tokens and repaying 3`] = `
[
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_ALLOWANCE",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "spenderAddress": "0xfakeSwapRouterContractAddress",
          "tokenAddress": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_BALANCES",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCES_ALL",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_POOLS",
      ],
    },
  ],
]
`;

exports[`useSwapTokensAndRepay > calls useSendTransaction with the correct parameters for swapping tokens and repaying full loan 3`] = `
[
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_BALANCE_OF",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "tokenAddress": "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_ALLOWANCE",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
          "spenderAddress": "0xfakeSwapRouterContractAddress",
          "tokenAddress": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_TOKEN_BALANCES",
        {
          "accountAddress": "0x3d759121234cd36F8124C21aFe1c6852d2bEd848",
          "chainId": 97,
        },
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_V_TOKEN_BALANCES_ALL",
      ],
    },
  ],
  [
    {
      "queryKey": [
        "GET_POOLS",
      ],
    },
  ],
]
`;
